src/argosim/metrics_utils.py

```diff
@@ -98,3 +98,198 @@ def compute_metrics(img1, img2):
+
+    Parameters
+    ----------
+    beam : np.ndarray
+        2D beam image.
+    fit_result : dict
+        Dictionary containing the ellipse parameters (center, width, height, angle_deg).
+    scale : float
+        Scale factor to enlarge or shrink the ellipse mask (default is 1.0).
+
+    Returns
+    -------
+    beam_masked : np.ndarray
+        Beam image with the main lobe masked (set to 0).
+    """
+    center_x, center_y = fit_result["center"]
+    width = fit_result["width"] * scale
+    height = fit_result["height"] * scale
+    angle = np.radians(fit_result["angle_deg"])
+
+    y, x = np.indices(beam.shape)
+    x_shifted = x - center_x
+    y_shifted = y - center_y
+
+    cos_angle = np.cos(angle)
+    sin_angle = np.sin(angle)
+    x_rot = cos_angle * x_shifted + sin_angle * y_shifted
+    y_rot = -sin_angle * x_shifted + cos_angle * y_shifted
+
+    mask = (x_rot**2 / (width / 2) ** 2 + y_rot**2 / (height / 2) ** 2) < 1
+
+    beam_masked = beam.copy()
+    beam_masked[mask] = 0
+    return beam_masked
+
+
+def compute_sll(beam, fit_result=None, scale=3):
+    """Compute sll
+
+    Compute the Side-Lobe Level (SLL) of a beam using an elliptical mask.
+
+    Parameters
+    ----------
+    beam : np.ndarray
+        The beam image (2D).
+    fit_result : dict
+        Elliptical fit result containing center, width, height, angle_deg.
+    scale : float
+        Scale factor for the ellipse mask.
+
+    Returns
+    -------
+    sll_db : float
+        Side-Lobe Level in dB.
+    """
+    if fit_result is None:
+        fit_result = fit_elliptical_beam(beam)
+    main_lobe_peak = np.max(np.abs(beam))
+    beam_masked = mask_main_lobe_elliptical(np.abs(beam), fit_result, scale=scale)
+    side_lobe_peak = np.max(beam_masked)
+    sll_db = 10 * np.log10(side_lobe_peak / main_lobe_peak + 1e-12)
+    return sll_db
+
+
+def compute_fwhm(beam, fit_result=None):
+    """Compute fwhm
+
+    Compute FWHM from beam
+
+    Parameters
+    ----------
+    beam : np.ndarray
+        The beam image (2D)
+    fit_result : dict
+        Dictionary containing the ellipse parameters (center, width, height, angle_deg).
+
+    Returns
+    -------
+    tuple
+        FWHM along x and y.
+    """
+    if fit_result is None:
+        fit_result = fit_elliptical_beam(beam)
+    return fit_result["width"], fit_result["height"]
+
+
+def compute_eccentricity(beam, fit_result=None):
+    """Compute eccentricity
+
+    Compute the eccentricity of an ellipse from beam.
+
+    Parameters
+    ----------
+    beam : np.ndarray
+        The beam image (2D)
+    fit_result : dict
+        Dictionary containing the ellipse parameters (center, width, height, angle_deg).
+
+    Returns
+    -------
+    float
+        Ellipticity e ∈ [0, 1], where 0 = circle and 1 = highly elongated.
+    """
+    if fit_result is None:
+        fit_result = fit_elliptical_beam(beam)
+    fwhm_a, fwhm_b = compute_fwhm(beam, fit_result)
+    A = max(fwhm_a, fwhm_b)
```

Comment:
In principle this is not necessary because the fit already gives you the larger axis first (eigenvalues are sorted by size). So you know that `fwhm_a >= fwhm_b`.